src/components/CodeInput/__test__/inputItems.spec.js

```diff
@@ -0,0 +1,74 @@
+import React from 'react';
+import { mount } from 'enzyme';
+import InputItems from '../inputItems';
+import InputItem from '../inputItem';
+
+describe('<InputItems />', () => {
+    it('should have 4 InputItem elements rendered when value array has a length of 4', () => {
+        const component = mount(<InputItems value={['', '', '', '']} />);
+        expect(component.find(InputItem).length).toBe(4);
+    });
+    it('should have isActive only on the focusedIndex input', () => {
+        const component = mount(<InputItems value={['1', '', '', '']} focusedIndex={1} />);
+        expect(
+            component
+                .find(InputItem)
+                .at(1)
+                .prop('isActive'),
+        ).toBe(true);
+    });
+    it('should have tabIndex passed only to the focusedIndex input', () => {
+        const tabIndex = 3;
+        const component = mount(
+            <InputItems value={['1', '', '', '']} focusedIndex={1} tabIndex={tabIndex} />,
+        );
+        expect(
+            component
+                .find(InputItem)
+                .at(1)
+                .prop('tabIndex'),
+        ).toBe(3);
+    });
+    it('should have the initial values properly assigned to each input', () => {
+        const values = ['1', '2', '3', '4'];
+        const component = mount(<InputItems value={values} />);
+        values.forEach((value, index) => {
+            expect(
+                component
+                    .find(InputItem)
+                    .at(index)
+                    .prop('value'),
+            ).toBe(value);
+        });
+    });
+    it('should have unique ids on each input identified by base id and input index', () => {
+        const values = ['', '', '', ''];
+        const id = 'code-input-base';
+        const component = mount(<InputItems value={values} id={id} />);
+        values.forEach((value, index) => {
+            const singleId = `${id}-${index}`;
```

Comment:
we should construct this id using our uniqueId hook instead of use the index since it will assure that the id will be unique, remember that id can't be twice in the entire web page

Reply:
For this case, we are using the uniqueID hook in the parent

`const inputId = useUniqueIdentifier('code-input');`

Then it is passed to InputItems which renders unique ids for inputs based on parentID and index, setting there a unique id for each input.

For my that solution work, if not for learning purposes if possible can you share why?

--

Reply:
test removed for now because we are removing id's over the inputs for now; until we fully integrate accesibility.